simulation/traffic_simulator/src/behavior/longitudinal_speed_planning.cpp

```diff
@@ -0,0 +1,326 @@
+      2 * (target_speed - current_twist.linear.x - current_accel.linear.x * acceleration_duration) /
+      (acceleration_duration * acceleration_duration);
+    ret.max_deceleration_rate =
+      2 * (target_speed - current_twist.linear.x - current_accel.linear.x * acceleration_duration) /
+      (acceleration_duration * acceleration_duration);
+  } else {
+    if (isAccelerating(target_speed, current_twist)) {
+      ret.max_acceleration = constraints.max_acceleration_rate * acceleration_duration -
+                             std::sqrt(
+                               (constraints.max_acceleration_rate * acceleration_duration *
+                                constraints.max_acceleration_rate * acceleration_duration) +
+                               2 * constraints.max_acceleration_rate *
+                                 (current_twist.linear.x +
+                                  current_accel.linear.x * acceleration_duration - target_speed)) +
+                             current_accel.linear.x;
+    } else {
+      ret.max_deceleration = -constraints.max_deceleration_rate * acceleration_duration +
+                             std::sqrt(
+                               (constraints.max_deceleration_rate * acceleration_duration *
+                                constraints.max_deceleration_rate * acceleration_duration) -
+                               2 * constraints.max_deceleration_rate *
+                                 (current_twist.linear.x -
+                                  current_accel.linear.x * acceleration_duration - target_speed)) +
+                             current_accel.linear.x;
+    }
+  }
+  return ret;
+}
+
+auto LongitudinalSpeedPlanner::getDynamicStates(
+  double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints & constraints,
+  const geometry_msgs::msg::Twist & current_twist,
+  const geometry_msgs::msg::Accel & current_accel) const
+  -> std::tuple<geometry_msgs::msg::Twist, geometry_msgs::msg::Accel, double>
+{
+  if (std::fabs(target_speed) > constraints.max_speed) {
+    THROW_SEMANTIC_ERROR(
+      "Target speed is ", std::to_string(target_speed), " , it overs ", entity,
+      "'s max_speed:", std::to_string(constraints.max_speed));
+  }
+  double linear_jerk = planLinearJerk(target_speed, constraints, current_twist, current_accel);
+  auto accel = forward(linear_jerk, current_accel, constraints);
+  auto twist = forward(accel, current_twist, constraints);
+  accel = timeDerivative(current_twist, twist);
+  linear_jerk = timeDerivative(current_accel, accel);
+  return std::make_tuple(twist, accel, linear_jerk);
+}
+
+auto LongitudinalSpeedPlanner::getRunningDistance(
+  double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints & constraints,
+  const geometry_msgs::msg::Twist & current_twist, const geometry_msgs::msg::Accel & current_accel,
+  double current_linear_jerk) const -> double
+{
+  constexpr double twist_tolerance = 0.01;
```

Comment:
Please provide a rationale for the value 0.01 as a comment (if there is no particular technical basis, write it as such)

Reply:
Add comment about it but this comment was not marked as updated.
Please check diff

Reply:
Your change does not resolve my point.
It's self-explanatory that 0.01 is a hard-coded value.
So your comment is not worth much.
What you have to explain is not what the value of 0.01 is, but how it was determined.